### DISCUSSION TITLE
Failed to download video

### DISCUSSION THREAD

**Discussion by P-Beshkov** [aclap-dev/video-downloadhelper#726] [Help and Q&A]
### Your question

Hi, video couldn't be downloaded, details below.

### Addon & CoApp version and details

{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://app.pluralsight.com/ilx/video-courses/2188277d-4004-4d29-858e-3827ad0932b7",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#1 @ 0000024a61f39440] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://app.pluralsight.com/video/manifests/eyJ0b2tlbiI6IjMyNmE0MjQzZDVhZjVjMzg4MGZhY2E0NDJkNGYxMjk2MzIzYzhiZjdlYWRkZGUwY2NjNjA2ZjI5MTA5NzczZWIiLCJleHBpcmF0aW9uIjoxNzE2ODU3NjY2LCJjZG4iOnsibmFtZSI6ImVkZ2VjYXN0IiwiYmFzZVVybCI6Imh0dHBzOi8vdmlkMjAucGx1cmFsc2lnaHQuY29tIn19/eae75894-6cb5-4db6-a41d-3411168528e2/2fb2d958-1d7c-4f89-bb65-8205d6b4ae30/hls_aac-96k-eng.m3u8.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "id": "coapp_failure_7bc4ea6e-8d8f-4a16-82e2-cb82350f2da4",
    "downloadable_id": "downloadable_7788994816535955"
  }
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.2.3a1 beta, CoApp 2.0.19.

I went to your web page & got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cad65c94-c223-4793-bf5a-03751f066137)

When VDH tried to download that video, it generated the same 403 Forbidden error you got.

So I tried to go to Plan B.  That image shows an HLS master manifest in the Network Monitor.  I tried to run ffprobe on it.  That also got 403 Forbidden.  In cases such as this, I always reload the page & rerun the ffprobe, trying to be quicker about it the second time.  These things sometimes expire.  But the 403 error was constant.

This amounts to strealth DRM.  They haven't actually imposed DRM protection on the video.  But they've put some kind of protection on the manifest & normal civilians like us don't know the magic spell to get into the manifest.  Bottom line: They have successfully protected this content against downloading.  Seems a bit extreme for a video that isn't even a minute & a half.  But there you go.
